mapper/orm/src/main/java/org/hibernate/search/mapper/orm/massindexing/impl/HibernateOrmMassIndexingContext.java

```diff
@@ -53,7 +61,8 @@ public HibernateOrmMassIndexingContext(HibernateOrmMassIndexingMappingContext ma
 	@Override
 	public <T> PojoMassIndexingLoadingStrategy<? super T, ?> loadingStrategy(PojoRawTypeIdentifier<T> expectedType) {
 		LoadingTypeContext<T> typeContext = typeContextProvider.forExactType( expectedType );
-		return new HibernateOrmMassIndexingLoadingStrategy<>( typeContext.loadingStrategy() );
+		return new HibernateOrmMassIndexingLoadingStrategy<>( typeContext.loadingStrategy(),
+				conditionalExpression( typeContext.jpaEntityName() ), typeContextProvider );
```

Comment:
Why don't you just pass the type context directly to `conditionalExpression`? You are essentially calling `typeContextProvider.forExactType( typeContextProvider.typeIdentifierForEntityName( typeContext.jpaEntityName() ) )`, which will return... `typeContext`.